rust/parquet/src/arrow/schema.rs

```diff
@@ -1194,6 +1372,17 @@ mod tests {
             });
     }
 
+    #[test]
+    #[should_panic(expected = "Parquet does not support writing empty structs")]
+    fn test_empty_struct_field() {
+        let arrow_fields = vec![Field::new("struct", DataType::Struct(vec![]), false)];
+        let arrow_schema = Schema::new(arrow_fields);
+        let converted_arrow_schema = arrow_to_parquet_schema(&arrow_schema);
+
+        assert!(converted_arrow_schema.is_err());
+        converted_arrow_schema.unwrap();
+    }
```

Comment:
Instead of `should_panic`, could we test the error type and message? `should_panic` is generally reserved to catch panics, not to raise the panic in the test itself.

Reply:
I'll fix what you've raised here in a follow up PR